src/Process/PointCloudProcessing.js

```diff
@@ -165,6 +165,7 @@ export default {
             if (elt.obj) {
                 elt.obj.material.visible = true;
                 elt.obj.material.size = layer.pointSize;
+                elt.obj.material.uniforms.mode.value = layer.mode;
```

Comment:
that would break using a material with no mode uniform.
you could instead set here  `elt.obj.material.mode = layer.mode;` and update the uniform inside the `updateUniforms` function.

Reply:
Good idea!

Reply:
Actually, what about passing `layer` to `updateUniforms`? That's more or less the pattern we used in iTowns: update uniforms according to some config values attached to layer. It'll make it explicit, and would avoid yet another value copy (they are getting difficult to follow).